From 6349c7350aa46ca46ab5e116444e32c9b45b044b Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 15 Dec 2015 04:56:49 -0800
Subject: [PATCH 0875/1767] UBUNTU: SAUCE: fixup: get rid of unused var build
 warning

Signed-off-by: John Johansen <john.johansen@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 security/apparmor/label.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index 0b3a80a..98a54d0 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -1769,9 +1769,6 @@ static struct aa_label *labelset_next_invalid(struct aa_labelset *ls)
 	read_lock_irqsave(&ls->lock, flags);
 
 	__labelset_for_each(ls, node) {
-		struct aa_profile *p;
-		struct label_it i;
-
 		label = rb_entry(node, struct aa_label, node);
 		if ((label_invalid(label) || vec_invalid(label->ent, label->size)) &&
 		    aa_get_label_not0(label))
@@ -1847,6 +1844,11 @@ static struct aa_label *__label_update(struct aa_label *label)
 			__aa_label_remove(labels_set(label), label, l);
 			write_unlock_irqrestore(&ls->lock, flags);
 			tmp = aa_label_insert(labels_set(l), l);
+			if (tmp != l) {
+				write_lock_irqsave(&ls->lock, flags);
+				__aa_update_replacedby(label, tmp);
+				write_unlock_irqrestore(&ls->lock, flags);
+			}
 			goto out;
 		}
 	} else
-- 
1.9.3

